test_shaders.py

```diff
@@ -459,19 +557,35 @@ def main():
     parser.add_argument('--opt',
             action = 'store_true',
             help = 'Run SPIRV-Tools optimization passes as well.')
+    parser.add_argument('--reflect',
+            action = 'store_true',
+            help = 'Test reflection backend.')
+    parser.add_argument('--parallel',
+            action = 'store_true',
+            help = 'Execute tests in parallel.  Useful for doing regression quickly, but bad for debugging and stat output.')
+    
+    global args;
     args = parser.parse_args()
-
     if not args.folder:
         sys.stderr.write('Need shader folder.\n')
         sys.exit(1)
 
+    if (args.parallel and (args.malisc or args.force_no_external_validation)):
```

Comment:
Should this be blocked for --update paths? Can the tempfile creation create race conditions?

Reply:
there's one process per input file, and each output file maps to exactly one input file, so there shouldn't be any chance of a race condition where two processes are trying to write to the same output.  However I can err on the side of caution.